### PR TITLE
Repair Spock

### PR DESCRIPTION
**Pull request by fudler**
Signed-off-by: Christian Oertel <christian.oertel@tngtech.com>

### CODE REVIEW

File: example-projects/spock/src/test/groovy/com/tngtech/jgiven/spock/stages/Given.groovy [hunk 1]
```diff
@@ -2,13 +2,13 @@ package com.tngtech.jgiven.spock.stages
...
-        self()
+    Given some_state() {
+        return self()
     }
 
     def some_state_$(String param) {
-        self()
+        return self()
     }
 }
```
Comment (l-1squared) [inline example-projects/spock/src/test/groovy/com/tngtech/jgiven/spock/stages/Given.groovy:14]: POSIX conformant line endings :D

File: example-projects/spock/build.gradle [hunk 2]
```diff
@@ -1,14 +1,28 @@
...
+    mavenCentral()
+    jcenter()
+}
+
 dependencies {
-    implementation 'org.spockframework:spock-core:1.3-groovy-2.5'
-    testImplementation 'com.tngtech.jgiven:jgiven-spock:latest.integration'
+    implementation "org.spockframework:spock-core:2.0-M4-groovy-3.0"
+    implementation "org.spockframework:spock-junit4:2.0-M4-groovy-3.0"
+    testImplementation "com.tngtech.jgiven:jgiven-spock:${jGivenVersion}"
```
Comment (l-1squared) [inline example-projects/spock/build.gradle:22]: how do the spock versions here relate to the one used by jgiven-spock?

Comment (l-1squared) [reply example-projects/spock/build.gradle:22]: oh, those are super old. But, on second thought. Don't touch them. Renovatebot should ask us to update them soon enough

**Review by l-1squared — APPROVED** [review]

File: example-projects/spock/build.gradle (hunk 2 above)
Comment (jsalinaspolo) [reply example-projects/spock/build.gradle:22]: I do not think that the current implementation of jgiven-spock is ready for spock2 that still not released

Comment (l-1squared) [reply example-projects/spock/build.gradle:22]: @jsalinaspolo  
Oh my. That is a serious concern that you raise here. 
And indeed,  I hadn't noticed that the "M4" in the package name stands for milestone. 
Yet, I have already used a different PR to update the jgiven-spock package using the same `org.spockframework:spock-junit4:2.0-M4-groovy-3.0` glue module that is used here. I will however, go and double check that this change is not breaking (it obviously did not break the tests) 
I asked @fudler to have a look into splitting the jgiven-spock package into a jgiven-spock1 and jgiven-spock2 version, something that you seem to have started roughly a year ago. 
Maybe it is indeed more advisable to keep the current package at spock1 and use a spock2 package to gradually build up to spock2.

**Pull request merged**